lib/Sema/CSApply.cpp

```diff
@@ -845,6 +845,17 @@ namespace {
       bool isUnboundInstanceMember =
         (!baseIsInstance && member->isInstanceMember());
 
+      // There are several shortcomings that prevent us from being able to
+      // support an unbound instance method reference with a 'super' base.
+      // For now, just diagnose it.
+      if (isSuper && isUnboundInstanceMember) {
+        context.Diags.diagnose(base->getLoc(),
+                               diag::super_instance_method_on_static_self,
+                               baseTy, DeclNameRef(member->getFullName()))
```

Comment:
Can you please make it a fix so we don't even try to apply an invalid (for now) solution? I think it could be added to `resolveOverload` method.